### DISCUSSION TITLE
Methods starting with "is" are ignored

### DISCUSSION THREAD

**Discussion by kevin-yu-0602** [victools/jsonschema-generator#443] [Q&A] [answered]
Hi there,
I am running into an issue where it seems like I am unable to discover methods that start with "is" and have corresponding fields that also start with "is" as they are getting ignored by the ignore checks wired in by the Jackson module.
The config I am using:
```java
        SchemaGeneratorConfigBuilder configBuilder =
                new SchemaGeneratorConfigBuilder(SchemaVersion.DRAFT_2019_09, OptionPreset.PLAIN_JSON)
                        .with(new JacksonModule(JacksonOption.INCLUDE_ONLY_JSONPROPERTY_ANNOTATED_METHODS));
        configBuilder
                .without(Option.NONPUBLIC_NONSTATIC_FIELDS_WITHOUT_GETTERS)
                .with(
                        Option.NONSTATIC_NONVOID_NONGETTER_METHODS,
                        Option.FIELDS_DERIVED_FROM_ARGUMENTFREE_METHODS);
        configBuilder.forMethods()
                .withPropertyNameOverrideResolver(method -> Optional
                        .ofNullable(method.getAnnotationConsideringFieldAndGetter(JsonProperty.class))
                        .map(JsonProperty::value)
                        .orElse(null));
```
The POJO I would like to generate a schema for:
```java
class BasicObject {
  private final boolean isVisible;
  private final boolean isHistoricallyInteresting;
  ...
  @JsonProperty("isVisible")
  public boolean isVisible() {
    return isVisible;
  }

  @JsonProperty("historicallyInteresting")
  public boolean isHistoricallyInteresting() {
    return isHistoricallyInteresting;
  }
}
```

Tracing through a debugger a bit, it seems like the `isVisible()` method is getting ignored because here we fetch the getter field
https://github.com/victools/jsonschema-generator/blob/main/jsonschema-module-jackson/src/main/java/com/github/victools/jsonschema/module/jackson/JacksonModule.java#L310

which has the name `isVisible` and which does not match with the internal name fetched here for the property, which is "visible".
https://github.com/victools/jsonschema-generator/blob/main/jsonschema-module-jackson/src/main/java/com/github/victools/jsonschema/module/jackson/JacksonModule.java#L297

It seems like once something is deemed to be ignored there is no way to make an exception here. Is there any way I can wire in special behavior to get out of this, thanks!

**Comment by CarstenWickner** [comment, marked as answer]
Hi @kevin-yu-0602,

your observation is correct. Inside the `JacksonModule`, when checking for a getter method's visibility, the underlying field is being considered too, which falls on your feet here it seems.
There is no `JacksonOption` for disabling this, but I made the corresponding methods `protected` so you may override them, e.g. like this:
```java
new JacksonModule(JacksonOption.INCLUDE_ONLY_JSONPROPERTY_ANNOTATED_METHODS) {
    @Override
    protected boolean shouldIgnoreField(FieldScope field) {
        // bypass undesired filtering based on fields; also affects getter methods that Jackson doesn't recognize as such
        // e.g., when the field and method name are the same
        return false;
    }
}
```
Adding this to your given example produces the (I assume to be) expected schema:
```json
{
  "$schema" : "https://json-schema.org/draft/2019-09/schema",
  "type" : "object",
  "properties" : {
    "historicallyInteresting" : {
      "type" : "boolean"
    },
    "isVisible" : {
      "type" : "boolean"
    }
  }
}
```

You can, of course, choose to re-do parts of what the original [`JacksonModule`](https://github.com/victools/jsonschema-generator/blob/main/jsonschema-module-jackson/src/main/java/com/github/victools/jsonschema/module/jackson/JacksonModule.java) does in that method.

----

Just for background: the `JacksonModule` of the schema generator tries to mimic the logic employed by Jackson when generating JSON payloads. In your example, the method `isVisible()` is not recognized as the conventional getter method for the `isVisible` field, i.e., the `isVisible` field is interpreted as not having a getter method and thereby gets ignored.

**Reply by kevin-yu-0602** [reply]
thanks @CarstenWickner your solution is concise and worked like a charm!
